doc/source/data/working-with-llms.rst

```diff
@@ -48,305 +57,292 @@ The processor expects input rows with a ``prompt`` field and outputs rows with b
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-If you encounter CUDA out of memory errors, Ray Data LLM provides several configuration options to optimize GPU memory usage:
+For multi-LoRA batch inference:
 
 .. literalinclude:: doc_code/working-with-llms/basic_llm_example.py
     :language: python
-    :start-after: __gpu_memory_config_example_start__
-    :end-before: __gpu_memory_config_example_end__
-
-**Key strategies for handling GPU memory issues:**
-
-- **Reduce batch size**: Start with smaller batches (8-16) and increase gradually
-- **Lower `max_num_batched_tokens`**: Reduce from 4096 to 2048 or 1024
-- **Decrease `max_model_len`**: Use shorter context lengths when possible
-- **Set `gpu_memory_utilization`**: Use 0.75-0.85 instead of default 0.90
-- **Use smaller models**: Consider using smaller model variants for resource-constrained environments
-
-If you run into CUDA out of memory, your batch size is likely too large. Set an explicit small batch size or use a smaller model, or a larger GPU.
+    :start-after: __lora_config_example_start__
+    :end-before: __lora_config_example_end__
 
-.. _model_cache:
+See :doc:`the vLLM with LoRA example</llm/examples/batch/vllm-with-lora>` for details.
 
-How to cache model weight to remote object storage
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Optimized model loading
+~~~~~~~~~~~~~~~~~~~~~~~
 
-While deploying Ray Data LLM to large scale clusters, model loading may be rate
-limited by HuggingFace. In this case, you can cache the model to remote object
-storage (AWS S3 or Google Cloud Storage) for more stable model loading.
+Use RunAI streamer for faster model loading:
 
-Ray Data LLM provides the following utility to help uploading models to remote object storage.
+.. note::
+    Install vLLM with runai dependencies: ``pip install -U "vllm[runai]>=0.10.1"``
 
-.. code-block:: bash
+.. literalinclude:: doc_code/working-with-llms/basic_llm_example.py
+    :language: python
+    :start-after: __runai_config_example_start__
+    :end-before: __runai_config_example_end__
 
-    # Download model from HuggingFace, and upload to GCS
-    python -m ray.llm.utils.upload_model \
-        --model-source facebook/opt-350m \
-        --bucket-uri gs://my-bucket/path/to/facebook-opt-350m
-    # Or upload a local custom model to S3
-    python -m ray.llm.utils.upload_model \
-        --model-source local/path/to/model \
-        --bucket-uri s3://my-bucket/path/to/model_name
+Serve deployments
+~~~~~~~~~~~~~~~~~
 
-And later you can use remote object store URI as `model_source` in the config.
+Share a vLLM engine across multiple processors using :ref:`Ray Serve <serving-llms>`:
 
 .. literalinclude:: doc_code/working-with-llms/basic_llm_example.py
     :language: python
-    :start-after: __s3_config_example_start__
-    :end-before: __s3_config_example_end__
+    :start-after: __shared_vllm_engine_config_example_start__
+    :end-before: __shared_vllm_engine_config_example_end__
```

Comment:
this is a pretty important pattern. i would be explicit to use key description words like multi-turn and agentic.